### ISSUE TITLE
Make callbacks more firendly

### ISSUE THREAD

**Issue by pompex** [UniSharp/laravel-filemanager#448]
I'm using your file manager along with the summernote. 

Depending on the file, I'd like to insert it as image, if its an image but in case its a file, I would like to create link.

Your current callback provides only two parameters:

- url to the file
- relative path to the file

I mean, instead of those two, why not simply pass the object with more details about the file? 

Take a look at my current callback:

```
window.SetUrl = function( url, path ) {
	var ext = url.split( '.' ).pop();

	if ( $.inArray( ext, ['jpg', 'jpeg', 'png', 'gif', 'svg', 'bmp'] ) !== -1 ) {
		context.invoke( 'insertImage', url );
		return;
	}

	context.invoke( 'createLink', {
		text: 'File',
		url: url,
		isNewWindow: true,
	} );
};
```

I would like to get file name for the url and it's extension and I have to do this with some additional code. Which is annoying.

Why not simply do this?

```
window.setURL = function(file) {
     // here we go
     file.name // name of the file
     file.url // url to the file
     file.path // relative path to the file
     file.mime // mime type of the file
     // maybe more of them like file.size? 
}
```

This is my suggestion. Would make life definately easier. 👍 

---

On the other hand: if you are using jQuery anyways - why don't you simply use jQuery triggers? This would make it a lot more functionable, since you could capture the event anywhere very easly and more than once.

**Comment by g0110280** [comment]
@pompex Thank you, these are great suggestions. I will put this in our 2.0 milestone.

**Comment by g0110280** [comment]
Implemented in `v2.0.0-alpha6`. Thanks @pompex .
